test/ETO/EquityToken.js

```diff
@@ -96,32 +99,113 @@ contract("EquityToken", ([admin, nominee, company, broker, ...holders]) => {
+      ).to.be.rejectedWith(EvmError);
+    });
 
     // cases for successful destroy, rejected due to controller, not enough balance etc.
     it("should destroy tokens");
 
-    it("should change token controller");
-
-    it("should change nominee");
-
     // should be a set of tests with different rounding, we should be able to run it on platform as well
     it("should convert equity token amount to shares");
 
-    it("should set token symbol and other metadata from eto terms correctly");
+    it("should set token symbol and other metadata from eto terms correctly", async () => {
+      const equityTokenName = await etoTerms.EQUITY_TOKEN_NAME();
+      const equityTokenSymbol = await etoTerms.EQUITY_TOKEN_SYMBOL();
+      const equityTokenShareNominalValue = await etoTerms.SHARE_NOMINAL_VALUE_EUR_ULPS();
+      const equityTokenPrecision = await tokenTerms.EQUITY_TOKENS_PRECISION();
+      const equityTokensPerShare = await tokenTerms.EQUITY_TOKENS_PER_SHARE();
+
+      expect(await equityToken.name()).to.have.string(equityTokenName);
+      expect(await equityToken.symbol()).to.have.string(equityTokenSymbol);
+      expect(await equityToken.decimals()).to.be.bignumber.eq(equityTokenPrecision);
+      expect(await equityToken.tokensPerShare()).to.be.bignumber.eq(equityTokensPerShare);
+      expect(await equityToken.shareNominalValueEurUlps()).to.be.bignumber.eq(
+        equityTokenShareNominalValue,
+      );
+    });
   });
 
   describe("IEquityTokenController tests", () => {
-    // extend implementation of TestNullEquityTokenController to be able to control all operations via mocks
-    it("can block transfers");
-    it("can block distribute");
-    it("can block issueTokens");
-    it("can block destroyTokens");
-    it("can block closeToken");
-    it("can block changing equity token controller");
-    it("can block changing nominee");
-    it("should change token controller");
-    it("should change nominee");
+    it("should change token controller", async () => {
+      const newEquityTokenController = await TestNullEquityTokenController.new(universe.address);
+
+      await equityToken.changeTokenController(newEquityTokenController.address);
+
+      expect(await equityToken.tokenController()).to.be.bignumber.eq(
+        newEquityTokenController.address,
+      );
+
+      // clean up
+      await equityToken.changeTokenController(equityTokenController.address);
+      expect(await equityToken.tokenController()).to.be.bignumber.eq(equityTokenController.address);
+    });
+
+    it("should change nominee", async () => {
+      const newNominee = holders[0];
+
+      await equityToken.changeNominee(newNominee, { from: company });
+
+      expect(await equityToken.nominee()).to.be.bignumber.eq(newNominee);
+
+      // clean up
+      await equityToken.changeNominee(nominee, { from: company });
+      expect(await equityToken.nominee()).to.be.bignumber.eq(nominee);
+    });
+
+    it("can block transfers", async () => {
```

Comment:
1. please also add tests for `transferFrom` and ERC223 transfers
2. add tests for approve() and `onApprove`
3. please add positive tests: can transfer, can approve etc.

why? because I want to use this code and write generic test suite for all tokens with controller so we need comprehensive test

Reply:
Ready. Please review once again.